dags/utils/s3_io.py

```
@@ -9,18 +9,22 @@
     # return an S3 client with the default region (or us-east-1 if not set)
     return boto3.client("s3", region_name=region)
 
 # upload bytes to an S3 bucket
 def put_bytes(bucket: str, key: str, data: bytes):
     # get an S3 client and upload the bytes to the bucket
-    s3_client().put_object(Bucket=bucket, Key=key, Body=data)
+
+    try:
+        s3_client().put_object(Bucket=bucket, Key=key, Body=data)
+    except ClientError as e:
+        raise RuntimeError(f"Failed to upload bytes to S3 bucket {bucket} with key {key}: {e}")
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)